prov/verbs/src/fi_verbs.c

```diff
@@ -844,6 +875,15 @@ fi_ibv_msg_ep_recvv(struct fid_ep *ep, const struct iovec *iov, void **desc,
 	return fi_ibv_msg_ep_recvmsg(ep, &msg, 0);
 }
 
+static inline int
+fi_ibv_post_send(struct ibv_qp *qp, struct ibv_send_wr *wr,
+		struct ibv_send_wr **bad_wr)
+{
+	int ret;
+	ret = ibv_post_send(qp, wr, bad_wr);
+	return (ret == ENOMEM) ? -FI_EAGAIN : -ret;
```

Comment:
The checks for ibv_post_send/recv should be against -1, with errno checked for ENOMEM. I.e.
if (ret)
    return (errno == ENOMEM) ? -FI_EAGAIN : -errno;
else
   return 0;